File: backend/app/schemas/arc.py
from pydantic import Field

from app.db.models import ResourcePressure
from app.schemas.common import APIModel


class ActRead(APIModel):
    name: str
    start_round: int = Field(ge=1)
    end_round: int = Field(ge=1)
    tone: str
    gm_instructions: str
    resource_pressure: ResourcePressure
    director_notes: str | None = None


class ArcRead(APIModel):
    name: str
    description: str | None = None
    acts: list[ActRead] = Field(min_length=1)

File: frontend/src/types/arc.ts
/** Resource pressure level for an act */
export type ResourcePressure = 'low' | 'medium' | 'high' | 'critical'

/** A single act within a narrative arc */
export interface Act {
  name: string
  startRound: number
  endRound: number
  tone: string
  gmInstructions: string
  resourcePressure: ResourcePressure
  directorNotes?: string
}

/** Player-defined narrative arc (matches shared/schemas/arc.json) */
export interface Arc {
  name: string
  description?: string
  acts: Act[]
}
